doc/src/guides/solving/solve-polynomial.md

```diff
@@ -0,0 +1,515 @@
+the multiplicity of roots:
+
+```py
+>>> solve(expression, x, dict=True)
+[{x: -2}, {x: 3}]
+>>> solve(symbolic, x, dict=True)
+[{x: -a}, {x: b}]
+```
+
+Refer to [](solve-equation-algebraically.md) for more about using
+{func}`~.solve`.
+
+### Algebraic Solution With Root Multiplicities
+
+#### `roots`
+
+{func}`~.roots` can give explicit expressions for the roots of polynomials that
+have symbolic coefficients (that is, if there are symbols in the coefficients)
+if {func}`~.factor` does not reveal them. However, it may fail for some
+polynomials. Here are examples of {func}`~.roots`:
+
+```py
+>>> roots(expression, x)
+{-2: 2, 3: 1}
+>>> roots(symbolic, x)
+{-a: 2, b: 1}
+```
+
+It returns results as a dictionary, where the key is the root (for example, -2)
+and the value is the multiplicity of that root (for example, 2).
+
+{func}`~.roots` function uses a combination of techniques (factorization,
+decomposition, radical formulae) to find expressions in radicals if possible for
+the roots. When it can find some radical expressions for the roots, it returns
+them along with their multiplicity. This function will fail for most high-degree
+polynomials (five or greater) because they do not have radical solutions, and
+there is no guarantee that they have closed-form solutions at all, as explained
+by the [Abel-Ruffini
+theorem](https://en.wikipedia.org/wiki/Abel%E2%80%93Ruffini_theorem).
+
+#### Factor the Equation
+
+A different approach is to factor a polynomial using {func}`~.factor`, which
+does not give the roots directly but can give you simpler expressions:
+
+```py
+>>> expression_expanded = expand(expression)
+>>> expression_expanded
+x**3 + x**2 - 8*x - 12
+>>> factor(expression_expanded)
+(x - 3)*(x + 2)**2
+>>> symbolic_expanded = expand(symbolic)
+>>> symbolic_expanded
+-a**2*b + a**2*x - 2*a*b*x + 2*a*x**2 - b*x**2 + x**3
+>>> factor(symbolic_expanded)
+(a + x)**2*(-b + x)
+```
+
+{func}`~.factor` can also factorize a polynomial in a given [polynomial
+ring](polys-ring) which can reveal roots lie in the coefficient ring. For
+example, if the polynomial has rational coefficients, then {func}`~.factor` will
+reveal any rational roots. If the coefficients are polynomials involving, for
+example, symbol $a$ with rational coefficients then any roots that are
+polynomial functions of $a$ with rational coefficients will be revealed. In this
+example, {func}`~.factor` reveals that $x = a^2$ and $x = -a^3 - a$ are roots:
+
+```py
+>>> from sympy import expand, factor
+>>> from sympy.abc import x, a
+>>> p = expand((x - a**2)*(x + a + a**3))
+>>> p
+-a**5 + a**3*x - a**3 - a**2*x + a*x + x**2
+>>> factor(p)
+(-a**2 + x)*(a**3 + a + x)
+```
+
+### Exact Numeric Solution With Root Multiplicities
+
+#### `real_roots`
+
+If the roots to your polynomial are real, using {func}`~.real_roots` ensures
+that only real (not complex or imaginary) roots will be returned.
+
+```py
+>>> real_roots(expression)
```

Comment:
`expression` is defined far above this example so it's not clear what it equals.

Reply:
Also, maybe it makes more sense to use an example here that has some non-real roots, like `real_roots(x**3 - 1)`.